src/drivers/driver_zephyr.c

```diff
@@ -1720,6 +1817,29 @@ static int _wpa_drv_zep_set_key(void *priv,
 static int wpa_drv_zep_set_key(void* priv,
 			       struct wpa_driver_set_key_params *params)
 {
+	struct zep_drv_if_ctx *if_ctx = NULL;
+	enum key_flag key_flag        = params->key_flag;
+
+	if_ctx = priv;
+
+	if (if_ctx->ft_roaming && (key_flag & KEY_FLAG_PAIRWISE)) {
+		return wpa_drv_zep_save_pairwise_key_params(priv, params);
```

Comment:
The issue itself looks generic enough but this fix is a hack, without using extended keys (storing both keys in parallel) the packet loss is common either in re-keying or roaming, do we know how the nl80211 handles this scenario? I can also check that and see if we can come up with a proper solution than storing and installing keys at a later point.

Reply:
I removed the fix in zephyr driver and added it to our wifi driver. If other vendor uses zephyr also has this issue, we can discuss more generic fix in zephyr driver in the future. Thanks!